### ISSUE TITLE
[C#] Implement new Concept Exercise: integral-numbers

### ISSUE THREAD

**Issue by NextNebula** [exercism/v3#780]
This issue describes how to implement the `integral-numbers` concept exercise for the C# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Integral Numbers in C#.

## Learning objectives

- Know of the difference between signed and unsigned integral types.
- Know of the existence of the integral types: `sbyte`, `byte`, `short`, `ushort`, `int`, `uint`, `long` and `ulong`.
- Know when to use which integral type.

## Out of scope

- BigInteger.
- Overflows.
- Nullable integral types.

## Concepts

This Concepts Exercise's Concepts are:

- `integral-numbers`: know of the difference between signed and unsigned integral types; know of the existence of the integral types: `sbyte`, `byte`, `short`, `ushort`, `int`, `uint`, `long` and `ulong`; know when to use which integral type.

## Prequisites

This Concept Exercise's prerequisites Concepts are:

- `numbers`: define numbers and apply arithmetic and logic to them.

### Hints

- [Integral numeric types][integral-numeric-types]: overview of the integral numeric types.
- [Numeric conversions][numeric-conversions]: overview of implicit and explicit numeric conversions.

### After

- [Integral numeric types][integral-numeric-types]: overview of the integral numeric types.
- [Numeric conversions][numeric-conversions]: overview of implicit and explicit numeric conversions.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/csharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/languages/csharp/reference/README.md#reference-docs
[reference-integer]: https://github.com/exercism/v3/blob/master/reference/types/integer.md
[reference-example]: https://github.com/exercism/v3/blob/master/reference/types/string.md#implementations
[analyzer]: https://github.com/exercism/csharp-analyzer
[representer]: https://github.com/exercism/csharp-representer
[exercise-example]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/numbers-floating-point
[design-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/design.md
[config.json-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/config.json
[concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[integral-numeric-types]: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/integral-numeric-types
[numeric-conversions]: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/numeric-conversions

**Comment by ErikSchierboom** [comment]
I just realized that it would be great if this exercise can also introduce the various forms of [integer literals](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/integral-numeric-types#integer-literals): normal, hexadecimal and binary. As well as introducing the digit separator (`_`) character.

**Comment by ErikSchierboom** [comment]
@EarthlingRich I've updated this issue to conform to the latest in Concept naming and issue contents. Progressive insight has suggested that also introducing overflows might be a bit much, so I've created [a separate exercise](https://github.com/exercism/v3/issues/1138) to cover that Concept.

Let me know if you have any questions and if you are still interested in working on this.

**Comment by ErikSchierboom** [comment]
This issue should not have been closed as it has not yet been implemented :)

**Comment by mikedamay** [comment]
@EarthlingRich please consider introducing primitive explicit casts here.  I think they fit better here than in `explicit-casts`.  Let me know what you think.

**Comment by mikedamay** [comment]
Hi @NextNebula.  I’m not sure if you’ve kept up with the v3 progress on the C# track but we’re in the process of finishing off a bunch of introductory exercises and it would great if we could include the `lists` exercise to which you are currently assigned.

If you think you will be in a position to work on this in the next couple of weeks then please let us know and we will set our expectations according.  If not, could you let us know and we’ll be happy to reassign the exercise.  We appreciate both good intentions and how life gets in the way so no worries.

If you have any uncommitted work that you think we could use or cannibalize then please let us know.

**Comment by mikedamay** [comment]
Hi @NextNebula.  I’m proposing to reassign this issue (following the above comment).  If you come across this in the next few days and want to take ownership again then please leave a comment.  In any case, I hope you have the opportunity to work on the project sometime soon.  It is very worth while.

**Comment by mikedamay** [comment]
closed by PR #2105

**Comment by ErikSchierboom** [comment]
Thanks for working on this @mikedamay!
